host1x-fence: follow kernel API changes

class_create() lost its owner argument in 6.4 and the devnode callback
takes a const struct device * since 6.2.

diff --git a/drivers/gpu/host1x-fence/dev.c b/drivers/gpu/host1x-fence/dev.c
index 2229978..b79683c 100644
--- a/drivers/gpu/host1x-fence/dev.c
+++ b/drivers/gpu/host1x-fence/dev.c
@@ -446,7 +446,7 @@ static const struct file_operations dev_file_fops = {
 #if defined(NV_CLASS_STRUCT_DEVNODE_HAS_CONST_DEV_ARG) /* Linux v6.2 */
 static char *host1x_fence_devnode(const struct device *dev, umode_t *mode)
 #else
-static char *host1x_fence_devnode(struct device *dev, umode_t *mode)
+static char *host1x_fence_devnode(const struct device *dev, umode_t *mode)
 #endif
 {
 	*mode = 0666;
@@ -465,7 +465,7 @@ static int host1x_uapi_init(struct host1x_uapi *uapi)
 #if defined(NV_CLASS_CREATE_HAS_NO_OWNER_ARG) /* Linux v6.4 */
 	uapi->class = class_create("host1x-fence");
 #else
-	uapi->class = class_create(THIS_MODULE, "host1x-fence");
+	uapi->class = class_create("host1x-fence");
 #endif
 	if (IS_ERR(uapi->class)) {
 		err = PTR_ERR(uapi->class);
